rados/ioctx.go

```diff
@@ -628,3 +628,13 @@ func (ioctx *IOContext) BreakLock(oid, name, client, cookie string) (int, error)
 		return int(ret), getError(ret)
 	}
 }
+
+// GetLastVersion will return the version number of the last object read or
+// written to.
+//
+// Implements:
+//  uint64_t rados_get_last_version(rados_ioctx_t io);
+func (ioctx *IOContext) GetLastVersion() uint64 {
+	v := C.rados_get_last_version(ioctx.ioctx)
```

Comment:
Should we add 
if (ioctx.ioctx == nil) {
     return 0, ErrNoIOContext
}
Your call because other functions of the file are not doing it.

Reply:
Sure.

Reply:
Err. I just realized this function doesn't return an error normally because this ceph call always just returns a value. Rather than complicating the interface to the function I think a panic in this case isn't the worst thing in the world.

Reply:
I don't really agree with that. In many other cases we prevent segfaults when consumers of the API make a mistake. Adding a check to validate that the `ioctx.ioctx != nil` is a minor enhancement, but benefits the stability of the API.

From my understanding of `src/librados/librados_c.cc`, passing `ioctx.ioctx == nil` will cause a crash:
```C++
extern "C" uint64_t _rados_get_last_version(rados_ioctx_t io)
{
  tracepoint(librados, rados_get_last_version_enter, io); 
  librados::IoCtxImpl *ctx = (librados::IoCtxImpl *)io;
  uint64_t retval = ctx->last_version();
  tracepoint(librados, rados_get_last_version_exit, retval);
  return retval;
}
```

Reply:
This is a tough one for me. I feel that adding an error return *just* to prevent api "abuse" just makes things more complicated for the callers. You pretty much have to create an invalid IOContext (like we do in the tests when we check these) to trigger these conditions. It's not as clumsy when the ceph api's already return an error to add a 3-line check, so I don't generally object to those.
That said it's not much work to add so if you feel strongly about this, let me know and I can add an "extra" error return to these and just learn to live with it.

Reply:
Yes, I feel strongly that we should prevent segfaults when we can. Even if the cause is incorrect use of the API.

It is a little surprising to see that the Ceph libraries do not check this...

Reply:
OK, done.